### PR TITLE
[netcdf-c] Fix dependency libmath

### LINKED ISSUE #12169: [netCDF] Fix dependency on embree (ParaView CI error)
@JackBoosY  The real culprit for the paraview CI error is the netCDF target. 
```
set_target_properties(netCDF::netcdf PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "${_IMPORT_PREFIX}/lib/libmath.a;ZLIB::ZLIB;hdf5::hdf5-static;hdf5::hdf5_hl-static;CURL::libcurl"
)
```
So the following needs fixing:
- add embree as a feature to netCDF and fix the build-depends. Deactivate the search for embree otherwise. 
- replace the direct library link with the embree maths target
- add the appropriate find_dependency call to netCDFConfig.cmake if build with the feature

### PR DESCRIPTION
**Pull request by JackBoosY**
Since `netcdf-c` requires the `math` library, the system in Linux also provides `libm`, but `embree3` also generates the `libmath` library. Due to search order issues, embree's libmath library will be used first.

So change the math library name order to prevent the use of embree's math library.

Fixes #12169

### CODE REVIEW

**Comment by Neumann-A** [comment]
No that is incorrect. netcdf-c searches for any usable math library. This might be either libmath.a or a system provided libm.a. As such the CMakeLists.txt needs to differentiate between those two a bit more and have a option like `USE_LIBMATH_INSTEAD_OF_LIBM/USE_EMBREE_LIBMATH` or similar. It is not a necessity to build netcdf-c with libmath. A system provided libm would also be fine.

**Comment by JackBoosY** [comment]
@Neumann-A Yes, the math library can be provided by the system, but not necessarily.
We cannot determine whether the math library exists in the build environment, so we must rely on embree to ensure a smooth build.

**Comment by Neumann-A** [comment]
@JackBoosY: A lot of ports would fail in CI if `-lm` wouldn't be available. 

If you change the order to 
`FIND_LIBRARY(HAVE_LIBM NAMES m libm math)`
or simply remove math than you should have what is intended in netcdf-c. I think that it picked up libmath.a from embree was more by accident than by design

**Comment by Neumann-A** [comment]
considering that the commit introducing the `find_library` call does not even have a commit message and was 8 years ago.

**Comment by JackBoosY** [comment]
@Neumann-A Should be okay now.

File: ports/netcdf-c/fix-dependency-libmath.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+     MESSAGE(FATAL_ERROR "Unable to find the math library.")
+diff --git a/netCDFConfig.cmake.in b/netCDFConfig.cmake.in
+index 102828f..c5052cc 100644
+--- a/netCDFConfig.cmake.in
++++ b/netCDFConfig.cmake.in
+@@ -16,6 +16,7 @@ include(CMakeFindDependencyMacro)
+ find_dependency(ZLIB)
+ find_dependency(HDF5 CONFIG)
+ find_dependency(CURL CONFIG)
++find_dependency(embree CONFIG)
```
Comment (ras0219) [inline ports/netcdf-c/fix-dependency-libmath.patch:22]: Why is this needed without adding `embree` to `Build-Depends`?

File: ports/netcdf-c/CONTROL [hunk 2]
```diff
@@ -1,5 +1,6 @@
 Source: netcdf-c
-Version: 4.7.3-1
+Version: 4.7.3
+Port-Version: 2
 Build-Depends: hdf5, curl
```
Comment (JackBoosY) [inline ports/netcdf-c/CONTROL:4]:
```suggestion
Build-Depends: hdf5, curl, embree3
```

File: ports/netcdf-c/fix-dependency-libmath.patch (hunk 1 above)
Comment (JackBoosY) [reply ports/netcdf-c/fix-dependency-libmath.patch:22]: Good catch, thanks!

File: ports/netcdf-c/CONTROL [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Source: netcdf-c
 Version: 4.7.3
-Port-Version: 3
-Build-Depends: hdf5, curl, embree3(linux|osx)
+Port-Version: 4
+Build-Depends: hdf5, curl, embree3
```
Comment (Neumann-A) [inline ports/netcdf-c/CONTROL:4]:
```suggestion
Build-Depends: hdf5, curl
```
a) on windows the search for embree is not performed
b) you switched the library search order so libmath and as such embree will always be last. The possibility that it gets picked up is near zero since most systems should have libm.

File: ports/netcdf-c/fix-dependency-libmath.patch (hunk 1 above)
Comment (Neumann-A) [inline ports/netcdf-c/fix-dependency-libmath.patch:22]:
```suggestion

```
would have never been needed even if libmath is linked due to it being linked by full path instead of a target!

Comment (JackBoosY) [reply ports/netcdf-c/fix-dependency-libmath.patch:22]: I forgot that...

**Review by PhoebeHui — APPROVED** [review]
